src/niamoto/cli/commands/deploy.py

```
@@ -0,0 +1,51 @@
+        subprocess.run(["git", "init"], check=True)
+        subprocess.run(["git", "checkout", "-b", branch], check=True)
+        subprocess.run(["git", "add", "."], check=True)
+        subprocess.run(["git", "commit", "-m", "Deploy to GitHub Pages"], check=True)
+        subprocess.run(["git", "remote", "add", "origin", repo], check=True)
+        subprocess.run(["git", "push", "-f", "origin", branch], check=True)
+
+        print_success(f"Deployed to GitHub Pages on branch: {branch}")
+
+    except Exception as e:
+        print_error(f"GitHub Pages deployment failed: {str(e)}")
+        raise click.Abort()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)